tests/integration/test_utils/data_source_config/postgres.py

```diff
@@ -0,0 +1,195 @@
+    type[JSONB],
+    # type[JSONPATH],
+    type[MACADDR],
+    # type[MACADDR8],
+    type[MONEY],
+    type[NUMERIC],
+    # type[NUMMULTIRANGE],
+    type[NUMRANGE],
+    type[OID],
+    type[REAL],
+    type[REGCLASS],
+    # type[REGCONFIG],
+    type[SMALLINT],
+    type[TEXT],
+    type[TIME],
+    type[TIMESTAMP],
+    # type[TSMULTIRANGE],
+    # type[TSQUERY],
+    # type[TSRANGE],
+    # type[TSTZMULTIRANGE],
+    type[TSTZRANGE],
+    type[TSVECTOR],
+    type[UUID],
+    type[VARCHAR],
+]
+
+
+class PostgreSQLDatasourceTestConfig(DataSourceTestConfig[PostgresColumnType]):
+    @property
+    @override
+    def label(self) -> str:
+        return "postgresql"
+
+    @property
+    @override
+    def pytest_mark(self) -> pytest.MarkDecorator:
+        return pytest.mark.postgresql
+
+    @override
+    def create_batch_setup(
+        self, data: pd.DataFrame, request: pytest.FixtureRequest
+    ) -> BatchTestSetup:
+        return PostgresBatchTestSetup(
+            data=data,
+            config=self,
+        )
+
+
+class PostgresBatchTestSetup(BatchTestSetup[PostgreSQLDatasourceTestConfig]):
+    def __init__(
+        self,
+        config: PostgreSQLDatasourceTestConfig,
+        data: pd.DataFrame,
+    ) -> None:
+        self.table_name = f"postgres_expectation_test_table_{randint(0, 1000000)}"
+        self.connection_string = "postgresql+psycopg2://postgres@localhost:5432/test_ci"
+        self.engine = create_engine(url=self.connection_string)
+        self.metadata = MetaData()
+        self.table: Union[Table, None] = None
+        self.schema = "public"
+        super().__init__(config=config, data=data)
+
+    @override
+    def make_batch(self) -> Batch:
+        name = self._random_resource_name()
+        return (
+            self._context.data_sources.add_postgres(
+                name=name, connection_string=self.connection_string
+            )
+            .add_table_asset(
+                name=name,
+                table_name=self.table_name,
+                schema_name=self.schema,
+            )
+            .add_batch_definition_whole_table(name=name)
+            .get_batch()
+        )
+
+    @override
+    def setup(self) -> None:
+        columns = [Column(name, type) for name, type in self.get_column_types().items()]
+        self.table = Table(self.table_name, self.metadata, *columns, schema=self.schema)
+        self.metadata.create_all(self.engine)
+        with self.engine.connect() as conn:
+            # pd.DataFrame(...).to_dict("index") returns a dictionary where the keys are the row
+            # index and the values are a dict of column names mapped to column values.
+            # Then we pass that list of dicts in as parameters to our insert statement.
+            #   INSERT INTO test_table (my_int_column, my_str_column) VALUES (?, ?)
+            #   [...] [('1', 'foo'), ('2', 'bar')]
+            conn.execute(insert(self.table), list(self.data.to_dict("index").values()))
```

Comment:
I'm getting deprecation warnings on this line. Nesting it under `with conn.begin():` resolves it. If we update to use that, it looks like on exiting a begin context manager, the transaction is committed for us.

Reply:
yeah that's because you're on sqla 1.4. Our tests run on 2.0 or this would fail (because of the warning), but i'll look into tweaking it to be compatible.

Reply:
fixed in [f13a6ba](https://github.com/great-expectations/great_expectations/pull/10582/commits/f13a6ba9f87718cc80dda1cc6149f2c72a928ec2)